Bring i18n and l10n features to Twinkle

Huji opened the pull request:
This one partially fixes #129 by removing the Twinkle.shared.standardList entirely from the module, moving it to the localization.

Huji commented on jquery-i18n.js:
```diff
@@ -0,0 +1,120 @@
+	};
+
+	$.i18n = i18n;
+})(jQuery);
```
@atlight I know you prefer _(...) over $.i18n._(...) but I don't know how to achieve that. Trying the following after the page loads:

`_ = $.i18n._`

And then calling `_('tw-core-portlet-name')` throws an error while `$.i18n._('tw-core-portlet-name')` of course works.

I cannot think of a way to get around the dependency of `_()` function on the entire `$.i18n` object.

atlight replied:
Yes, I saw that. It's rather odd. I'm sure there is a way around it using `apply()` or one of those weird functions. As I said, I'll look later.

Huji replied:
I have given you write access to my fork, so feel free to experiment there.

MusikAnimal replied:
Sorry, I accidentally deleted @Huji's comment (meant to delete my own). It read: `I have given you write access to my fork, so feel free to experiment there.`

Huji commented on modules/twinklel10n.js:
```diff
@@ -0,0 +1,84 @@
+
+// Localizing twinkle is also easy, for example:
+switch( mw.config.get('wgUserLanguage') ) {
+	case 'en':
```
@atlight This is an important part of the change that I somehow had managed not to push (accidentally put it in a separate branch).

As you can see, the default set of templates for 'en' is what EN WP uses. We can decide if we want to include any of the other translations (I vote no), but for now I am keeping a mock-translation for DE, just so you can try the demo link and see how it works (only two templates will be listed in DE interface).

Huji commented on modules/friendlyshared.js:
```diff
@@ -16,16 +16,16 @@
 	}
 };
 
 Twinkle.shared.callback = function friendlysharedCallback() {
```
@atlight In future, we can move this into the l10n file as well. Just like how we move the Twinkle.shared.standardLIst to there.

Huji commented:
@atlight sorry for being away from this for a while.

I cleaned up the code again. I also changed `$.i18n` to `$._` which is closer to what you liked. What are the next steps?

siddharthvp commented:
This PR has been superseded by other work as y'all are aware. Closing to avoid any confusion.

Huji commented:
Totally agree.
